cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/UnmanagedClusterInventory.test.tsx

```diff
@@ -1,17 +1,142 @@
 // React imports
 import React from 'react';
-import { BrowserRouter } from 'react-router-dom';
 
 // Library imports
-import { render } from '@testing-library/react';
-import '@testing-library/jest-dom';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/lib/node';
 
 // App imports
 import UnmanagedClusterInventory from './UnmanagedClusterInventory';
 
-describe('UnmanagedClusterLanding', () => {
-    test('should render', () => {
-        const view = render(<UnmanagedClusterInventory />, { wrapper: BrowserRouter });
-        expect(view).toBeDefined();
+jest.mock('react-router-dom', () => ({
+    ...(jest.requireActual('react-router-dom') as any),
+    useNavigate: () => jest.fn(),
+}));
+
+describe('UnmanagedClusterInventory component', () => {
+    const server = setupServer(
+        rest.get('/api/unmanaged', (req, res, ctx) => {
+            return res(
+                ctx.status(200),
+                ctx.json([
+                    {
+                        name: 'work-space-cluster',
+                        provider: 'MiniKube',
+                        status: 'Running',
+                    },
+                    {
+                        name: 'tanzu-cluster',
+                        provider: 'Kind',
+                        status: 'Stopped',
+                    },
+                    {
+                        name: 'ui-cluster',
+                        provider: 'Kind',
+                        status: 'Unknown',
+                    },
+                ])
+            );
+        })
+    );
+
+    beforeAll(() => server.listen({ onUnhandledRequest: 'bypass' }));
+    afterEach(() => server.resetHandlers());
+    afterAll(() => server.close());
+
+    const clickFirstDeleteButton = async function () {
+        const deleteBtns = await screen.findAllByText('Delete');
+        const firstDeleteBtn = deleteBtns[0];
+        fireEvent.click(firstDeleteBtn);
+    };
+
+    test('should render', async () => {
+        const view = render(<UnmanagedClusterInventory />);
+        await waitFor(() => {
+            expect(view).toBeDefined();
+        });
+    });
+
+    test('should always display a button to create an unmanaged cluster', async () => {
+        render(<UnmanagedClusterInventory />);
+
+        expect(await screen.findByText('create unmanaged cluster')).toBeInTheDocument();
+    });
+
+    test('should display three unmanaged cluster cards when unmanaged clusters are present', async () => {
+        render(<UnmanagedClusterInventory />);
+        const unmanagedClusterCards = await screen.findAllByTestId('unmanaged-cluster-card');
+        expect(unmanagedClusterCards.length).toBe(3);
```

Comment:
The number 3 is a little magical here; if you have the test data array defined above, then you could use the size of the array, and it would be clear why you expect that many cards